docs/sandboxing/README.md

```diff
@@ -10,9 +10,82 @@ There is no clear solution to this yet, but it should take heavy inspiration fro
 
 ### Linux
 
-We could use something like flatpak here, though its unlikely, the current most likely solution is to use linux namespaces to work like a docker-like system.
-
-This is mainly solved other then implementation details
+Linux offers different kinds of isolation using namespacing.
+A fairly low-level tool to access these capabilities is `unshare` that allows creating new namespaces for child processes.
+`unshare` is fine, if full-isolation should be achieved, but we definitely need *some* access to the host system, and without integrating with multiple tools, `unshare` by itself is undesirable.
+A popular tool that utilizes the same capabilities as `unshare` is `bubblewrap`, which is most prominently used by Flatpak. It also allows creating new namespaces, just like `unshare` but additionally provides easy ways to bind paths from the host into the mount namespace.
+
+#### Bubblewrap
+
+Using `bubblewrap` is quite simple. It can be configured by running the `bwrap` binary and defining command line arguments.
+As the name suggests it acts as a wrapper, so after specifying bwrap options the main program inside the sandbox can be specified.
+
+##### Base configuration
+
+- `--unshare-all --share-net` - Try to create all kinds of namespaces except for networking
+  - Doing unprivileged networking in userspace is slow, which is why we will NOT sandbox the root networking namespace
+- `--die-with-parent` - Kill sandbox if launcher crashes/exits
+- `--dev /dev --dev-bind-try /dev/dri /dev/dri` - Allow access to DRM devices for hardware accelerated rendering
+- `--proc /proc` - Mount namespaced procfs at /proc
+- `--unsetenv DBUS_SESSION_BUS_ADDRESS` - The sandbox doesn't have access to the dbus daemon anyway
+- Read-only bind mounts
+  - `/etc`
```

Comment:
Possibly dangerous ability for servers, can bruteforce passwords in `/etc/shadow` or get internal IPs from `/etc/hosts`

Reply:
It could probably be more granular. We definitely need `/etc/resolv.conf`. Not sure what else is needed

Reply:
If anything besides root has access to your /etc/shadow or you're running your stuff as root I have severe concerns... (and if your hash function is really that weak)
but other than that, a blanket allow for /etc still doesn't seem like the best approach

Reply:
`/etc/hosts` is needed

Reply:
should be more granular now